### ISSUE TITLE
BOINC Screen Saver does not turn the display off at the right time, per Windows Power Management setting to turn off the display

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#1058]
**Reported by JacobKlein on 6 Apr 41257338 14:13 UTC**
I have Windows 7 x64.

I believe there is a miscalculation somewhere in the BOINC Screensaver,  which may be preventing the display from being turned off at the right time, per the Windows Power Management setting.

When I have the Blank Screensaver set to wait 1 minute before starting  (via the Windows 7 screensaver settings), and then I have Windows Power  Management set to turn off the display after 2 minutes, the Blank Screensaver will start after 1 minute, and then a minute later, the monitor will go into amber power save mode. Thus it takes 2 minutes of no activity to turn off the display.

When I have the BOINC Screensaver set to wait 1 minute before starting (via the Windows 7 screensaver settings), and then I have Windows Power Management set to turn off the display after 2 minutes, the BOINC Screensaver will start after 1 minute, but then it will run for 2 minutes before the monitor will go into amber power save mode. Thus it takes 3 minutes of no activity to turn off the display.

It should have only taken 2 minutes of no activity to turn off the display.

Migrated-From: http://boinc.berkeley.edu/trac/ticket/1082

**Comment by romw** [comment]
**Commented by JacobKlein on 28 May 41277707 12:26 UTC**
Further testing shows that, if the !ScreenSaver is set to start after 1 minute, and settings are to change projects every minute, and Windows Power Management is set to turn off the display after 2 minutes... then the display never turns off!

So, I think there must be a timeout timer that's being incorrectly reset, any time a graphics app change occurs within the !ScreenSaver.

Feel free to email with questions.

**Comment by romw** [comment]
**Commented by JacobKlein on 19 Mar 43282437 08:17 UTC**
Emails discussing the issue again, during testing of 7.0.62:

----
> From: boinc_alpha [ On[[BR](mailto:boinc_alpha-bounces@ssl.berkeley.edu])] > Behalf Of Jacob Klein[> Sent: Thursday, April 11, 2013 3:41 AM[[BR]([BR]])] > To: BOINC Alpha[> Subject: [boinc_alpha]([BR]]) BOINC Screen Saver - Still does not turn the[> display off at the right time, per Windows Power Management setting to[[BR]([BR]])] > turn off the display[> [[BR]([BR]])] > I'm re-reporting this bug (Ticket !#1082 , opened 2 years ago) to the[> list...[[BR]([BR]])] > and reporting it as a Major Bug in the 7.0.60 test results, despite the[> issue also happening with prior versions.[[BR]([BR]])] > [> http://boinc.berkeley.edu/trac/ticket/1082  [[BR]([BR]])] > [> I call it major because the user has an expectation that the monitor[[BR]([BR]])] > will turn off (to save energy and save computation cycles)... when in[> reality, it does not work properly.[[BR]([BR]])] > [> The original bug report said:[[BR]([BR]])] > If the !ScreenSaver  is set to start after 1 minute, and settings are to[> change projects every minute, and Windows Power Management is set to[[BR]([BR]])] > turn off the display after 2 minutes... then the display never turns[> off! [[BR]([BR]])] > [> I just tested that again, on 7.0.62, in Windows 8 x64, and for 5[[BR]([BR]])] > minutes, the BOINC Screen Saver was still running. The monitor should[> have turned off after 2.[[BR]([BR]])] > Can you guys please (finally) fix this? You're wasting electricity and[> computation cycles, and ignoring the user's setting! [[BR]([BR]])] > [> Staring at the screen for minute after minute, stopwatch in hand,[[BR]([BR]])] > waiting for things to happen, like a good tester... then not getting the[> results...[[BR]([BR]])] > It's very frustrating. I hope you can fix it.[> [[BR]([BR]])] > Thanks,[> Jacob

----
[[BR]([BR]])]>  Subject: RE: [BOINC Screen Saver - Still does not turn the  display off at the right time, per Windows Power Management setting to  turn off the display[[BR](boinc_alpha])]> Date: Fri, 12 Apr 2013 11:26:29 -0400[From: [mailto:romw@romwnet.org]([BR]]>)[To: [mailto:jacob_w_klein@msn.com]([BR]]>); [[[BR](mailto:boinc_alpha@ssl.berkeley.edu][[BR]]>)]> What value do you have the 'blank screensaver' set too?[[[BR]([BR]]>)]> Is it before the monitor's power settings say it should go to sleep?[[[BR]([BR]]>)]> ----- Rom

----
'''From:''' Jacob Klein [[[BR](mailto:jacob_w_klein@msn.com])]'''Sent:''' Friday, April 12, 2013 11:52 AM[Rom Walton; BOINC Alpha[[BR]([BR]]'''To:''')]'''Subject:'''  RE: [BOINC Screen Saver - Still does not turn the display  off at the right time, per Windows Power Management setting to turn off  the display

I don't use the "blank screensaver" setting within the BOINC Screensaver.[[BR](boinc_alpha])][here is some more info that may help you to solve the problem:[[BR]([BR]]However,)](Note: Be sure you have several active tasks that show graphics)[screensaver after: Never[[BR]([BR]][[BR]]Blank)]Run BOINC screensaver for: 1[project screensavers for: 15[[BR]([BR]]Run)]Switch between project screensavers every: **5**[Power Options -> Turn off the display: 2 minutes[[BR]([BR]]Windows)]RESULT: The display **WILL** eventually turn off.[screensaver after: Never[[BR]([BR]][[BR]]Blank)]Run BOINC screensaver for: 1[project screensavers for: 15[[BR]([BR]]Run)]Switch between project screensavers every: **1**[Power Options -> Turn off the display: 2 minutes[[BR]([BR]]Windows)]RESULT: The display **WILL NOT** turn off! [seems that, when a new task's graphics gets loaded, it resets the timer that's used to turn the display off.[[BR]([BR]][[BR]]It)]That's what it looks like to me, at least.[this help?

----
Subject:  RE: [boinc_alpha]([BR]][[BR]]Does) BOINC Screen Saver - Still does not turn the display  off at the right time, per Windows Power Management setting to turn off  the display[Fri, 12 Apr 2013 12:17:39 -0400[[BR]([BR]]Date:)]From: [[mailto:jacob_w_klein@msn.com](mailto:romw@romwnet.org][[BR]]To:); [the screensaver will probably do the same thing.

IIRC,  just the act of starting a new process that uses OpenGL resets the  Windows timer for the display.  It isnt something we can work around.

I  suppose the long term fix would be to get Windows to tell us when the  monitor is supposed to shutdown and use that value as the blank time  when the blank time is greater.

----
'''From:''' Jacob Klein ![mailto:jacob_w_klein@msn.com](mailto:boinc_alpha@ssl.berkeley.edu]

Blanking) [Friday, April 12, 2013 12:42 PM[[BR]([BR]]'''Sent:''')]'''To:''' Rom Walton; BOINC Alpha[ RE: [boinc_alpha]([BR]]'''Subject:''') BOINC Screen Saver - Still does not turn the display  off at the right time, per Windows Power Management setting to turn off  the display

Blanking the screen is not the same as the display turning off.[user has set "Turn off the display" in Windows Power Options, and has a reasonable expectation that it will be honored.[[BR]([BR]][[BR]]The)]The long term fix would be to find a way to not reset the timer, thus honoring the Windows setting that was set by the user.[understand if there's nothing you can do for now.[[BR]([BR]][[BR]]I)]It's just an issue that was frustrating when I reported it (over 2 years ago), and is still frustrating today (as I retest it).[hope it can be given some shred of priority at some point.[[BR]([BR]]I)][RE: [boinc_alpha]([BR]]Jacob

----
[[BR]]Subject:) BOINC Screen Saver - Still does not turn the  display off at the right time, per Windows Power Management setting to  turn off the display[Fri, 12 Apr 2013 12:48:47 -0400[[BR]([BR]]Date:)]From: romw@romwnet.org[jacob_w_klein@msn.com; boinc_alpha@ssl.berkeley.edu[[BR]([BR]]To:)]

Blanking the screensaver prevents it from launching graphics applications  Thereby preventing the timer from being reset

----
[jacob_w_klein@msn.com[[BR]([BR]]From:)]To: romw@romwnet.org; boinc_alpha@ssl.berkeley.edu[ RE: [boinc_alpha]([BR]]Subject:) BOINC Screen Saver - Still does not turn the display  off at the right time, per Windows Power Management setting to turn off  the display[Fri, 12 Apr 2013 13:41:53 -0400[[BR]([BR]]Date:)][  So... let's start with a scenario, which I think would be a common scenario:[[BR]([BR]])]- User sets screensaver to start after 1 minute of user inactivity[User wants to show project graphics, switching every minute or two[[BR]([BR]]-)]- User also wants the display to turn off eventually, maybe after 15 minutes[your proposed changes:[[BR]([BR]][[BR]]With)]- (Measured from screensaver start) How long would project graphics show?[(Measured from screensaver start) When would the display go blank?[[BR]([BR]]-)]- (Measured from beginning of user inactivity) When would the display turn off?

**Comment by romw** [comment]
**Commented by JacobKlein on 6 Apr 43846622 21:07 UTC**
> From: jacob_w_klein@msn.com[To: boinc_alpha@ssl.berkeley.edu[[BR]([BR]]>)]> Date: Mon, 4 Nov 2013 17:17:06 -0500[Subject: [boinc_alpha]([BR]]>) Windows 8.1 - BOINC Screen Saver does not transition to turning off the display[[[BR]([BR]]>)]> I don't usually test the Screen Saver, but I did today.[It turns out, the previous Screen Saver bugs have only gotten worse under Windows 8.1.[[BR]([BR]]>)]> [ Basically, despite having "Turn off the display" set, if the BOINC  Screen Saver had been running, I cannot get the display to turn off  after any length of time.[[BR]([BR]]>)]> So, the monitor stays on (light  green), the BOINC Screen Saver stops showing stuff (presumably because  it thinks it doesn't need to anymore), and....[I'm left with a fully powered black screen.[[BR]([BR]]>)]> [When I tested using the "3D Text" Screen Saver, that Screen Saver worked how I expected.[[BR]([BR]]>)]> [I don't consider this a showstopper.[[BR]([BR]]>)]> But it does mean that Ticket !#1082 has gotten even worse under Windows 8.1, and monitors now cannot sleep.[Rom: I'd love to troubleshoot this with you to get it working like I'd expect.[[BR]([BR]]>)]> [Regards,[[BR]([BR]]>)]> Jacob

**Comment by Ageless93** [comment]
Is this still an issue?

**Comment by SteveScott** [comment]
I am having trouble with my monitor not shutting off when BOINC is active.

**Comment by reyo** [comment]
Still a thing.
I'm worried about my screen burn in as my computer is on 24/7/365.

Is there a legit workaround?

I'm not using BOINC screensaver and I'm running as service.
